src/main/java/gregtech/common/metatileentities/miner/Miner.java

```diff
@@ -0,0 +1,70 @@
+package gregtech.common.metatileentities.miner;
+
+import gregtech.client.model.miningpipe.MiningPipeModel;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import javax.annotation.Nonnull;
+
+public interface Miner {
+
+    /**
+     * Try to drain all mining resources required for one operation. (e.g. energy, mining fluids)
+     *
+     * @param minedBlockType Type of the block mined
+     * @param pipeExtended   Whether the pipe length got extended
+     * @param simulate       If {@code true}, this action will not affect the state of the game
+     * @return Whether the action was successful
+     */
+    boolean drainMiningResources(@Nonnull MinedBlockType minedBlockType, boolean pipeExtended, boolean simulate);
```

Comment:
The boolean for `pipeExtended` does not seem to be used in any of the implementations of this method. Was it left over from an old implementation?